internal/services/controller/controller.go

```diff
@@ -259,19 +272,49 @@ func (c *Controller) Run(ctx context.Context) error {
 }
 
 func (c *Controller) startConditionalInformersWithWatcher(ctx context.Context, conditionalInformers []conditionalInformer) {
+	tryConditionalInformers := conditionalInformers
 	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Minute*2, func(ctx context.Context) (done bool, err error) {
-		apiResourceLists := fetchApiResourceLists(c.discovery, c.log)
+		apiResourceLists := fetchAPIResourceLists(c.discovery, c.log)
 		if apiResourceLists == nil {
 			return false, nil
 		}
+		c.log.Infof("Cluster API server is available, trying to start conditional informers")
 
-		c.log.Infof("Cluster API server is available, starting conditional informers")
-		for _, informer := range conditionalInformers {
-			apiResourceListForGroupVersion := getApiResourceListByGroupVersion(informer.groupVersion, apiResourceLists)
-			if isResourceAvailable(informer.apiType, apiResourceListForGroupVersion) {
-				custominformers.NewHandledInformer(c.log, c.queue, informer.informer, informer.apiType, nil)
+		for _, informer := range tryConditionalInformers {
+			if informer.isApplied {
+				continue
 			}
+			apiResourceListForGroupVersion := getAPIResourceListByGroupVersion(informer.groupVersion, apiResourceLists)
+
+			resourceAvailable := isResourceAvailable(informer.apiType, apiResourceListForGroupVersion)
+			informerHasAccess := c.informerHasAccess(ctx, informer)
```

Comment:
We should fail fast and if the resource isn't even available, there's no reason to check for access.